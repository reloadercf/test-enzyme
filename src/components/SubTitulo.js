import React,{useState} from 'react';

function SubTitulo() {
    let [texto,setTexto]=useState({
        print:true,
        text:'Las habburguesas de don cangrejo'
    })
    let handleSetFalse=()=>{
        setTexto({...texto,print:false})
    }

    return (
        <div>
            {texto.print && <h2>{texto.text}</h2>}
            <button onClick={()=>handleSetFalse()}>Borrar hamburguesa</button>
        </div>
    );
}

export default SubTitulo;