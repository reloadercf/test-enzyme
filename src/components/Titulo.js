import React from 'react';

function Titulo({eltitulo}) {
    return (
        <div>
            <h1>{eltitulo}</h1>
        </div>
    );
}

export default Titulo;