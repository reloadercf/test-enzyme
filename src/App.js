import React from 'react';
import './App.css';
import Titulo from './components/Titulo'
import SubTitulo from './components/SubTitulo'

function App() {
  return (
    <div className="App">
      <Titulo eltitulo="Un titulo normal" />
      <SubTitulo />
      <p>Alg√∫n cambio</p>
    </div>
  );
}

export default App;
