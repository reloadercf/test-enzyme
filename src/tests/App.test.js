import React from 'react';
import {shallow} from 'enzyme'

import App from '../App';
import Titulo from '../components/Titulo'
import SubTitulo from '../components/SubTitulo'

describe("Test App",()=>{
  test("snapshot App",()=>{
    let elcomponente=shallow(<App />)
    expect(elcomponente).toMatchSnapshot();
  })
  test("Test H1 to <Titulo/>",()=>{
    let eltitulo="Burger queen pro"
    let elcomponente=shallow(<Titulo eltitulo={eltitulo}/>)
    let buscarH1=elcomponente.find("h1").html()
    //console.log(buscarH1)
    expect(buscarH1).toBe(`<h1>${eltitulo}</h1>`)
  })

  test("SubTitle TRUE",()=>{
    let elcomponente=shallow(<SubTitulo/>)
    expect(elcomponente.find("h2").exists()).toBe(true)
  })

  test("SubTitle False",()=>{
    let elcomponente=shallow(<SubTitulo/>)
    //let boton=elcomponente.find('button').simulate("click")
    elcomponente.find('button').simulate("click")
    expect(elcomponente.find("h2").exists()).toBe(false)
  })
})