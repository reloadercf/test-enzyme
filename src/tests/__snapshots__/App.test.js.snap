// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App snapshot App 1`] = `
<div
  className="App"
>
  <Titulo
    eltitulo="Un titulo normal"
  />
  <SubTitulo />
  <p>
    Alg√∫n cambio
  </p>
</div>
`;
